templates/types/llamaindexserver/fastapi/gen_ui.py

```diff
@@ -0,0 +1,363 @@
+                const aggregateEvents = () => {
+                    // code for aggregating events here
+                }
+
+                // styles for the component
+                const styles = {
+                    // styles for the component
+                }
+
+                // State for the component
+                // e.g: const [state, setState] = React.useState({});
+                // handle the state here
+
+                return (
+                    <div style={styles.container}>
+                        // UI code here
+                    </div>
+                )
+            }
+            // Don't need to export the component
+        ```
+    """
+
+    def __init__(self, llm: LLM, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.llm = llm
+
+    @step
+    async def start(self, ctx: Context, ev: StartEvent) -> RunWorkflowEvent:
+        workflow_input = ev.workflow_input
+        if not workflow_input:
+            raise ValueError("workflow_input is required")
+        events = ev.events
+        if not events:
+            raise ValueError(
+                "events is required, provide list of events that are emitted from the workflow that you want to generate the UI components for"
+            )
+        output_file = ev.output_file
+        if not output_file:
+            raise ValueError(
+                "output_file is required. Provide the path of the file to save the generated UI component"
+            )
+        await ctx.set("output_file", output_file)
+        return RunWorkflowEvent(
+            workflow_input=workflow_input,
+            events=events,
+        )
+
+    @step
+    async def run_workflow(
+        self, ctx: Context, ev: RunWorkflowEvent
+    ) -> WriteAggregationEvent:
+        """
+        Run the workflow to get the events.
+        """
+        from app.workflow import create_workflow
+
+        workflow = create_workflow()
+        print(
+            f"Running workflow with input: {ev.workflow_input}\n Waiting for the workflow to finish..."
+        )
+        handler = workflow.run(user_msg=ev.workflow_input)
+
+        workflow_events = []
+        async for event in handler.stream_events():
+            workflow_events.append(event)
+        await handler
+        print("Workflow finished")
+        selected_events = []
+        expected_event_names = [
+            event if isinstance(event, str) else event.__name__ for event in ev.events
+        ]
+        for event in workflow_events:
+            if type(event).__name__ in expected_event_names:
+                selected_events.append(event)
+            elif hasattr(event, "data"):
+                if event.data.__class__.__name__ in expected_event_names:
+                    selected_events.append(event.data)
+        if len(selected_events) == 0:
+            raise ValueError(f"No event {ev.events} found in the workflow events")
+        await ctx.set("events", selected_events)
```

Comment:
i would move this transformation (workflow code) -> to events schema outside of the workflow and start the workflow with an array of JSON schemas - also produce just the code string as output - reason:
a) then the workflow structure will be the same with TS
b) we can for a test easily call it with a pydantic test model (e.g. the one from the issue)